src/custom_gates.jl

```diff
@@ -18,6 +30,92 @@ function matrix_rep(g::DoubleExcitation)
     return SMatrix{16,16,ComplexF64}(mat)
 end
 
+@doc raw"""
+    DoubleExcitationPlus(ϕ)
+
+Generate the matrix representation of the [DoubleExcitationPlus](https://docs.pennylane.ai/en/stable/code/api/pennylane.DoubleExcitationPlus.html) gate.
+
+This gate performs an SO(2) rotation in the subspace {|1100⟩, |0011⟩} with a phase-shift on other states:
+
+```math
+|0011\rangle & \rightarrow \cos\left(\frac{\phi}{2}\right)|0011\rangle - \sin\left(\frac{\phi}{2}\right)|1100\rangle \\
+|1100\rangle & \rightarrow \cos\left(\frac{\phi}{2}\right)|1100\rangle + \sin\left(\frac{\phi}{2}\right)|0011\rangle \\
+|x\rangle & \rightarrow e^{\frac{i\phi}{2}}|x\rangle \quad \text{for all other basis states } |x\rangle
+```
+"""
+struct DoubleExcitationPlus <: AngledGate{1}
+    angle::NTuple{1,Union{Float64,FreeParameter}}
+    DoubleExcitationPlus(angle::T) where {T<:NTuple{1,Union{Float64,FreeParameter}}} =
+        new(angle)
+end
+Braket.chars(::Type{DoubleExcitationPlus}) = "G2+(ang)"
+Braket.qubit_count(::Type{DoubleExcitationPlus}) = 4
+Base.inv(g::DoubleExcitationPlus) = DoubleExcitationPlus(-g.angle[1])
+function matrix_rep(g::DoubleExcitationPlus)
+    cosϕ = cos(g.angle[1] / 2.0)
+    sinϕ = sin(g.angle[1] / 2.0)
+    eiϕ2 = exp(im * g.angle[1] / 2.0)
+    mat = diagm(ones(ComplexF64, 16))
```

Comment:
Why not just multiply by `eiϕ2` here and zero out the relevant rows later?

Reply:
Hope this looks good:
```
function matrix_rep(g::DoubleExcitationPlus)
    cosϕ = cos(g.angle[1] / 2.0)
    sinϕ = sin(g.angle[1] / 2.0)
    eiϕ2 = exp(im * g.angle[1] / 2.0)
    mat = diagm(eiϕ2 * ones(ComplexF64, 16))
    mat[4, :] .= 0
    mat[:, 4] .= 0
    mat[13, :] .= 0
    mat[:, 13] .= 0
    # Apply phase-shift to states outside rotation subspace
    mat[4, 4]   = cosϕ
    mat[13, 13] = cosϕ
    mat[4, 13]  = -sinϕ
    mat[13, 4]  = sinϕ
    return SMatrix{16, 16, ComplexF64}(mat)
end
```